src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1217,6 +1217,10 @@ private function registerAnnotationsConfiguration(array $config, ContainerBuilde
             throw new LogicException('Annotations cannot be enabled as the Doctrine Annotation library is not installed.');
         }
 
+        if (!class_exists('Doctrine\Common\Cache\ApcCache')) {
+            throw new LogicException('Annotations cannot be enabled as the Doctrine Cache library is not installed.');
```

Comment:
this exception must be inside the `if ('none' !== $config['cache']) {`. We don't need the doctrine/cache library when disabling the annotation caching.